src/utils.rs

```diff
@@ -11,26 +10,23 @@ pub fn distance(a: Coords2D, b: Coords2D) -> f64 {
 
 //Helper function to recreate path once goal is located
 pub fn rewind(start: &Node, closed: &Vec<Node>) -> Vec<Coords2D> {
-    let mut path = Vec::new();
+    let mut path = Vec::with_capacity(closed.len().pow(2) + 1);
```

Comment:
Not sure if this sizing is correct, but creating the Vec with a capacity helps reducing memory reallocations.

I don't fully understand the purpose of this function based on it's code, but for how it's used I believe it should only be sorting the closed nodes, shouldn't it? If that's the case, you could pass the closed Vec as owned, modify it with `.sort_by()` and then consume the vec with `.into_iter().map().collect()` to convert the soreted `Vec<Node>` into the return value of `Vec<Coords2D>`. If you need to discard some closed nodes, you can `.filter()` them out before the `.map()`. I'm happy to give you more insight on how to achieve this if you explain me the logic behind this function

Reply:
It should be sorting the closed nodes, but also creating new intermediate nodes between them. As you've probably seen, every node that is checked is added to the closed list, then once the the destination is found it needs to filter the closed list to only contain those nodes which are on the final route.

JPS is an optimization of A* which checks nodes 1 at a time, so for A* every node in the final route will automatically be on the closed list. In JPS, it jumps ahead, so it also needs to interpolate the nodes in between the jumps. The function tries to recreate the path with the aid of the parent field and do this interpolation as it goes.

Hopefully that makes sense?